### PR TITLE
make electron runtime deps optional

### LINKED ISSUE #2758: Don't distribute electron
It causes issues like https://github.com/theia-ide/theia/issues/1699 and https://github.com/theia-ide/theia/issues/2693. I suggest that we move `electron` to `devDependency` for `core` and `cli` that we still can use its api, but it will be up to the client to install it. For `cli` probably we can remove it altogether, also need to check whether `electron-builder` can cause similar issues.

We will need to fix https://github.com/theia-ide/theia/issues/775 first.

### LINKED ISSUE #4871: native-keymap should be optional dep
Motivated by https://github.com/theia-ide/theia-apps/issues/163

All next images are broken again 🙈 

cc @spoenemann

### PR DESCRIPTION
**Pull request by akosyakov**
fix #4871
fix #2758

### CODE REVIEW

File: dev-packages/electron/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "url": "https://github.com/theia-ide/theia.git"
+  },
+  "bugs": {
+    "url": "https://github.com/theia-ide/theia/issues"
+  },
+  "homepage": "https://github.com/theia-ide/theia",
+  "dependencies": {
+    "electron": "^3.1.7",
+    "electron-store": "^2.0.0",
+    "native-keymap": "^1.2.5"
```
Comment (kittaakos) [inline dev-packages/electron/package.json]: Please add [`fix-path`](https://github.com/theia-ide/theia/pull/3534/commits/381ff3942b951d6cce9ea26fd937433cd355a9ad) too.

**Comment by akosyakov** [comment]
It's ready for the review. Guys please have a look we need to fix theia-apps next images.

**Comment by akosyakov** [comment]
thanks @kittaakos, waiting for travis build and merging if no objections

**Comment by kittaakos** [comment]
> merging if no objections

Sure, if I understood the issue, the theia-apps are broken anyway so that it won't be worse.

**Comment by akosyakov** [comment]
@kittaakos it will allow to fix them, i more worry about people who build electron products that we break them now :)

File: dev-packages/application-package/src/application-package.ts [hunk 2]
```diff
@@ -46,6 +48,14 @@ export class ApplicationPackage {
         this.projectPath = options.projectPath;
         this.log = options.log || console.log.bind(console);
         this.error = options.error || console.error.bind(console);
+        if (this.isElectron()) {
+            const { version } = require('../package.json');
+            try {
+                require('@theia/electron/package.json');
+            } catch {
+                console.warn(`please install @theia/electorn@${version} as a runtime dependency`);
```
Comment (spoenemann) [inline dev-packages/application-package/src/application-package.ts]: Typo: `@theia/electron`

Comment (akosyakov) [reply dev-packages/application-package/src/application-package.ts]: Could you open a PR? :)

Comment (spoenemann) [reply dev-packages/application-package/src/application-package.ts]: #4878

**Pull request merged**